das/store_signing.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/binary"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/offchainlabs/nitro/arbstate"
+)
+
+var uniquifyingPrefix = []byte("Arbitrum Nitro DAS API Store:")
+
+type DasSigner func([]byte) ([]byte, error)
+
+func DasSignerFromPrivateKey(privateKey *ecdsa.PrivateKey) DasSigner {
+	return func(data []byte) ([]byte, error) {
+		return crypto.Sign(data, privateKey)
+	}
+}
+
+func applyDasSigner(signer DasSigner, data []byte, timeout uint64) ([]byte, error) {
+	return signer(dasStoreHash(data, timeout))
+}
+
+func DasRecoverSigner(data []byte, timeout uint64, sig []byte) (common.Address, error) {
+	pk, err := crypto.SigToPub(dasStoreHash(data, timeout), sig)
+	if err != nil {
+		return common.Address{}, err
+	}
+	return crypto.PubkeyToAddress(*pk), nil
+}
+
+func dasStoreHash(data []byte, timeout uint64) []byte {
+	var buf8 [8]byte
+	binary.BigEndian.PutUint64(buf8[:], timeout)
+	return crypto.Keccak256(uniquifyingPrefix, buf8[:], data)
+}
+
+type StoreSigningDAS struct {
+	inner  DataAvailabilityService
+	signer DasSigner
+}
+
+func NewStoreSigningDAS(inner DataAvailabilityService, signer DasSigner) DataAvailabilityService {
+	return &StoreSigningDAS{inner, signer}
+}
+
+func (s *StoreSigningDAS) Retrieve(ctx context.Context, cert *arbstate.DataAvailabilityCertificate) ([]byte, error) {
+	return s.inner.Retrieve(ctx, cert)
+}
+
+func (s *StoreSigningDAS) KeysetFromHash(ctx context.Context, ksHash []byte) ([]byte, error) {
+	return s.inner.KeysetFromHash(ctx, ksHash)
+}
+
+func (s *StoreSigningDAS) CurrentKeysetBytes(ctx context.Context) ([]byte, error) {
+	return s.inner.CurrentKeysetBytes(ctx)
+}
```

Comment:
Not necessary to change for this PR and you may prefer the way you have it currently for clarity, but if you changed StoreSigningDAS like so: 
```
type StoreSigningDAS struct {
    DataAvailabilityService
    signer DasSigner
}
```
Then it would automatically forward through all the methods you don't override.

Reply:
Nice!  I made this change.